store/sqlstore/user_store.go

```diff
@@ -326,28 +327,41 @@ func (us SqlUserStore) UpdateMfaActive(userId string, active bool) error {
 }
 
 // GetMany returns a list of users for the provided list of ids
-func (us SqlUserStore) GetMany(ids []string) ([]*model.User, error) {
+func (us SqlUserStore) GetMany(ctx context.Context, ids []string) ([]*model.User, error) {
 	query := us.usersQuery.Where(sq.Eq{"Id": ids})
 	queryString, args, err := query.ToSql()
 	if err != nil {
 		return nil, errors.Wrap(err, "users_get_many_tosql")
 	}
 
+	var db *gorp.DbMap
+	if hasMaster(ctx) {
+		db = us.GetMaster()
+	} else {
+		db = us.GetReplica()
+	}
```

Comment:
Should this become a function?

Reply:
Totally, I am just refraining from too much DRY right at this moment, since we have just started to do this. But it's clear that this needs to be refactored. I will send another PR after this.